Sources/Internal/OperationsManager.swift

```diff
@@ -28,34 +40,52 @@ class OperationsManager {
                                        .Type,
                                      publisher: ResultsPublisherType = .auto)
     -> any ObservableQueryRef {
-    switch publisher {
-    case .auto, .observableMacro:
-      if #available(iOS 17, macOS 14, tvOS 17, watchOS 10, *) {
-        return QueryRefObservation<ResultDataType, VariableType>(
-          request: request,
-          dataType: resultType,
-          grpcClient: self.grpcClient
-        ) as (any ObservableQueryRef)
-      } else {
-        return QueryRefObservableObject<ResultDataType, VariableType>(
-          request: request,
-          dataType: resultType,
-          grpcClient: grpcClient
-        ) as (any ObservableQueryRef)
+    queryRefAccessQ.sync {
```

Comment:
Can you add an inline comment here about why the work is delegated to a DispatchQueue? At a glace it seems to be pointless since the code is calling sync which (presumably) blocks anyways.

Reply:
Its a synchronized access to the underlying dictionary. The queue is being used as a means of serializing access to it.  Its a pattern used instead of locks.